typed-builder-macro/src/struct_info.rs

```diff
@@ -298,7 +319,11 @@ impl<'a> StructInfo<'a> {
             let body = &transform.body;
             (quote!(#(#params),*), quote!({ #body }))
         } else if field.builder_attr.setter.strip_option.is_some() {
-            (quote!(#field_name: #arg_type), quote!(Some(#arg_expr)))
+            if options_was_stripped {
```

Comment:
Nit: we can check this variable instead of the `field.builder_attr.setter.strip_option.is_some()` in the line above. If `strip_option` is `None`, the flag will be false anyway and we won't wrap `arg_expr` in a `Some`.